common/lib/xmodule/xmodule/x_module.py

```diff
@@ -1568,7 +1571,8 @@ def ajax_url(self):
         return self.handler_url(self.xmodule_instance, 'xmodule_handler', '', '').rstrip('/?')
 
     def get_block(self, block_id):
-        return self.get_module(self.descriptor_runtime.get_block(block_id))
+        block = self.descriptor_runtime.get_block(block_id)
+        return self.get_module(block) if block else None
```

Comment:
However, resetting `_child_instances` is also causing an infinite recursion in `TestXmlCoursesLoad`.

https://build.testeng.edx.org/job/edx-platform-all-tests-pr/1180/SHARD=1,TEST_SUITE=unit,label_exp=jenkins-worker/consoleFull

```
edx.courseware: ERROR: Error in index view: user=test, course=CourseDescriptorWithMixins(CombinedSystem(<lms.djangoapps.lms_xblock.runtime.LmsModuleSystem object at 0x44513c10>, <xmodule.modulestore.xml.ImportSystem object at 0x194f1b90>), <lms.djangoapps.lms_xblock.field_data.LmsFieldData object at 0x44513b90>, ScopeIds(user_id=2, block_type='course', def_id=BlockUsageLocator(SlashSeparatedCourseKey('edX', 'toy', '2012_Fall', None, None), 'course', '2012_Fall'), usage_id=BlockUsageLocator(SlashSeparatedCourseKey('edX', 'toy', '2012_Fall', None, None), 'course', '2012_Fall'))), chapter=Overview section=Toy_Videos position=3
15:20:07    Traceback (most recent call last):
15:20:07      File "/home/jenkins/workspace/edx-platform-all-tests-pr/SHARD/1/TEST_SUITE/unit/label_exp/jenkins-worker/lms/djangoapps/courseware/views.py", line 405, in _index_bulk_op
15:20:07        has_content = course.has_children_at_depth(CONTENT_DEPTH)
15:20:07      File "/home/jenkins/workspace/edx-platform-all-tests-pr/SHARD/1/TEST_SUITE/unit/label_exp/jenkins-worker/common/lib/xmodule/xmodule/x_module.py", line 378, in has_children_at_depth
15:20:07        return any(child.has_children_at_depth(depth - 1) for child in self.get_children())
15:20:07      File "/home/jenkins/workspace/edx-platform-all-tests-pr/SHARD/1/TEST_SUITE/unit/label_exp/jenkins-worker/common/lib/xmodule/xmodule/x_module.py", line 378, in <genexpr>
15:20:07        return any(child.has_children_at_depth(depth - 1) for child in self.get_children())
15:20:07      File "/home/jenkins/workspace/edx-platform-all-tests-pr/SHARD/1/TEST_SUITE/unit/label_exp/jenkins-worker/common/lib/xmodule/xmodule/x_module.py", line 378, in has_children_at_depth
15:20:07        return any(child.has_children_at_depth(depth - 1) for child in self.get_children())
15:20:07      File "/home/jenkins/workspace/edx-platform-all-tests-pr/SHARD/1/TEST_SUITE/unit/label_exp/jenkins-worker/common/lib/xmodule/xmodule/x_module.py", line 378, in <genexpr>
15:20:07        return any(child.has_children_at_depth(depth - 1) for child in self.get_children())
15:20:07      File "/home/jenkins/workspace/edx-platform-all-tests-pr/SHARD/1/TEST_SUITE/unit/label_exp/jenkins-worker/common/lib/xmodule/xmodule/x_module.py", line 376, in has_children_at_depth
15:20:07        return bool(self.get_children())
15:20:07      File "/home/jenkins/workspace/edx-platform-all-tests-pr/SHARD/1/TEST_SUITE/unit/label_exp/jenkins-worker/common/lib/xmodule/xmodule/x_module.py", line 416, in get_children
15:20:07        child = self.runtime.get_block(child_loc)
15:20:07      File "/home/jenkins/workspace/edx-platform-all-tests-pr/SHARD/1/TEST_SUITE/unit/label_exp/jenkins-worker/common/lib/xmodule/xmodule/x_module.py", line 1574, in get_block
15:20:07        block = self.descriptor_runtime.get_block(block_id)
15:20:07      File "/home/jenkins/workspace/edx-platform-all-tests-pr/SHARD/1/TEST_SUITE/unit/label_exp/jenkins-worker/common/lib/xmodule/xmodule/x_module.py", line 1574, in get_block
15:20:07        block = self.descriptor_runtime.get_block(block_id)
15:20:07      File "/home/jenkins/workspace/edx-platform-all-tests-pr/SHARD/1/TEST_SUITE/unit/label_exp/jenkins-worker/common/lib/xmodule/xmodule/x_module.py", line 1574, in get_block
```

Can we just use `get_display_items` for now?

Can we just use `get_display_items` then?